### DISCUSSION TITLE
How does Renovate determine PR Title, or Commit Message?

### DISCUSSION THREAD

**Discussion by gkzz** [renovatebot/renovate#17274] [Request Help] [answered]
Hello, all.
My question is how Renovate determines PR Title, or Commit Message.

If `:semanticPrefixFixDepsChoreOthers` is enabled, and some Semantic Commits are detected, Renovate decides `fix` or `chore` as the following says.
https://docs.renovatebot.com/presets-default/#semanticprefixfixdepschoreothers 

Another question is what the default values of commit message, PR title or PR description.

> Change the default values for branch name, commit message, PR title or PR description¶
You can use the branchName, commitMessage, prTitle or prBody configuration options to change the defaults for those settings.

https://docs.renovatebot.com/faq/#change-the-default-values-for-branch-name-commit-message-pr-title-or-pr-description

**Comment by rarkins** [comment, marked as answer]
It would take a long time to explain every aspect of commit messages and PR titles, so you need to be more specific, or read the docs and source code yourself. It would be like you posted to the Apache open source and asked "How does Apache answer requests?".

You can start with https://docs.renovatebot.com/configuration-options/#commitmessage
